controllers/managedocs_controller.go

```diff
@@ -913,6 +977,42 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+func getPrometheusWebSpec() *promv1.WebSpec {
+	locSecretRef := corev1.LocalObjectReference{
+		Name: prometheusServiceName,
+	}
+	keySelector := corev1.SecretKeySelector{
+		LocalObjectReference: locSecretRef,
+		Key:                  "tls.key",
+	}
+	certSelector := corev1.SecretKeySelector{
+		LocalObjectReference: locSecretRef,
+		Key:                  "tls.crt",
+	}
+	caBundleSelector := corev1.ConfigMapKeySelector{
+		LocalObjectReference: corev1.LocalObjectReference{
+			Name: caBundleConfigMapName,
+		},
+		Key: "service-ca.crt",
+	}
+
+	// configure TLS
+	webSpec := &promv1.WebSpec{
+		TLSConfig: &promv1.WebTLSConfig{
+			KeySecret: keySelector,
+			Cert: promv1.SecretOrConfigMap{
+				Secret: &certSelector,
+			},
+			ClientAuthType: fmt.Sprint(tls.VerifyClientCertIfGiven),
+			ClientCA: promv1.SecretOrConfigMap{
+				ConfigMap: &caBundleSelector,
+			},
+		},
+	}
```

Comment:
Most of these are constant values, pelase move into the prom template. 
And in the same breath, there is no need for `getPrometheusWebSpec`

Reply:
We should not do that. These are constants which are in package `controllers` and if we move `getPrometheusWebSpec()` function into `templates` package we have to import `controllers` package which then creates a cyclic dep, which is a bad practice.
If we create a separate package for the common vars and constants (say `defaults`), we can do this.

Reply:
The controller pkg has visibility to all public constants of the template package. 
Just more the constants there

We already did that for other templates in the past

Reply:
Moved the `getPrometheusWebSpec()` to templates.
Changed it to `PrometheusWebSpecTemplate`.